src/libotutil/ot-variant-utils.c

```diff
@@ -319,7 +319,7 @@ ot_variant_bsearch_str (GVariant   *array,
                         int        *out_pos)
 {
   gsize imax, imin;
-  gsize imid;
+  gsize imid = -1;
```

Comment:
GCC is unfortunately not smart enough to figure out that the loop is guaranteed to run at least once. Maybe if we put it an the else block of the `n == 0` condition, but this is easier.